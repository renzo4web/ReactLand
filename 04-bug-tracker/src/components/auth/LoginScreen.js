import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import validator from 'validator';
import { Link } from 'react-router-dom';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';
import { removeError, setError } from '../../actions/ui';

const LoginScreen = () => {
    const initialForm = {
        email: '',
        password: '',
    };
    const { msgError, loading } = useSelector((state) => state.ui);
    const dispatch = useDispatch();
    const [formValues, handleInputChange] = useForm(initialForm);

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        if (isFormValid()) {
            dispatch(startLoginEmailPassword(email, password));
        }
    };

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    };

    const isFormValid = () => {
        if (!validator.isEmail(email)) {
            dispatch(setError('Email is not valid'));
            return false;
        }

        if (password.length < 5) {
            dispatch(
                setError(
                    'Password should be at least 5 characters long and match each other'
                )
            );
            return false;
        }

        dispatch(removeError());
        return true;
    };

    return (
        <>
            <h1>Login</h1>

            {msgError && (
                <strong className='ui__alert-input'>{msgError}</strong>
            )}
            <form onSubmit={handleLogin}>
                <div className='ui__input-group'>
                    <label htmlFor='email'>Email</label>
                    <input
                        value={email}
                        placeholder='test@gmail.com'
                        onChange={handleInputChange}
                        name='email'
                        id='email'
                        type='text'
                    />
                </div>

                <div className='ui__input-group'>
                    <label htmlFor='password'>Password</label>
                    <input
                        value={password}
                        onChange={handleInputChange}
                        name='password'
                        id='password'
                        type='password'
                    />
                </div>

                <button disabled={loading} type='submit' className='auth__btn'>
                    Login
                </button>

                <hr />

                <div>
                    <p>Login with social networks</p>

                    <div onClick={handleGoogleLogin} className='google-btn'>
                        <div className='google-icon-wrapper'>
                            <img
                                className='google-icon'
                                src='https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg'
                                alt='google button'
                            />
                        </div>
                        <p className='btn-text'>
                            <b>Sign in with google</b>
                        </p>
                    </div>
                </div>

                <Link to='/auth/register'>Create new account</Link>
            </form>
        </>
    );
};

export default LoginScreen;
