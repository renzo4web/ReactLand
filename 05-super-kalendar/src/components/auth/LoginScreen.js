import React from 'react';
import Login from './Login';
import Register from './Register';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

const LoginScreen = () => {
	return (
		<Container className="my-auto">
			<Row
				className="my-auto bg-primary text-white p-5 rounded"
				as="main"
			>
				<Col sm>
					<Login />
				</Col>
				<Col sm>
					<Register />
				</Col>
			</Row>
		</Container>
	);
};

export default LoginScreen;
