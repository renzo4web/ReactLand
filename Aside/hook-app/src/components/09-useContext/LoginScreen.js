import React from "react";

const LoginScreen = () => {
  return (
    <div>
      <h1>Login</h1>
      <form action>
        <label htmlFor="name">
          Name
          <input type="text" />
        </label>
        <label htmlFor="password">
          Password
          <input type="password" />
        </label>
      </form>
    </div>
  );
};

export default LoginScreen;
