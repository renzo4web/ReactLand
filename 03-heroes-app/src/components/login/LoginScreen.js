import {
  Button,
  Heading,
  Center,
  Flex,
  Image,
  Box,
  Input,
  Text,
} from '@chakra-ui/react';
import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useAuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';
import img from './../../assets/heroes/dc-arrow.jpg';

const LoginScreen = () => {
  const [, dispatch] = useAuthContext();
  const [name, setName] = useState('');
  const history = useHistory();
  const lastPath =
    JSON.parse(localStorage.getItem('lastPath')) || '/';

  const handleLogin = () => {
    dispatch({
      type: types.login,
      payload: {
        name,
      },
    });
    history.replace(lastPath);
  };

  return (
    <Center h="100vh">
      <Box>
        <Flex direction="column">
          <Heading mb="20px" as="h1">
            Login
          </Heading>
          <Image
            my="1em"
            boxSize="200px"
            objectFit="cover"
            src={img}
            alt="Arrow"
          />
          <Text>Name:</Text>
          <Input
            size="md"
            placeholder="Mr Marvel"
            value={name}
            onChange={({ target }) => setName(target.value)}
            onBlur={({ target }) => setName(target.value)}
            mb="1em"
          />
          <Button
            onClick={() => handleLogin()}
            colorScheme="teal"
          >
            Login
          </Button>
        </Flex>
      </Box>
    </Center>
  );
};

export default LoginScreen;
