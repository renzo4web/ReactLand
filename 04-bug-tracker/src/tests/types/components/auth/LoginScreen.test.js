import configureStore from 'redux-mock-store'; //ES6 modules
import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import LoginScreen from '../../../../components/auth/LoginScreen';
import { MemoryRouter } from 'react-router-dom';
import {
    startGoogleLogin,
    startLoginEmailPassword,
} from '../../../../actions/auth';

jest.mock('../../../../actions/auth', () => ({
    startGoogleLogin: jest.fn(),
    startLoginEmailPassword: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: 'testinguid',
        name: 'Test',
    },
    bugs: {
        bugs: [],
        active: null,
    },
    ui: {
        msgError: null,
        loading: true,
    },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <LoginScreen />
        </MemoryRouter>
    </Provider>
);

describe('Test <LoginScreen />', () => {
    beforeEach(() => {
        store = mockStore(initState);
        jest.clearAllMocks();
    });

    test('wrapper should snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should trigger action startGoogleLogin', () => {
        wrapper.find('.google-btn').prop('onClick')();

        expect(startGoogleLogin).toHaveBeenCalled();
    });

    test('should trigger startLoginEmailPassword()', () => {
        const testEmail = 'test@gmail.com';
        const testPassword = 'cccccccccccccc';
        wrapper
            .find('input#email')
            .simulate('change', {
                target: { name: 'email', value: testEmail },
            });
        wrapper
            .find('input#password')
            .simulate('change', {
                target: { name: 'password', value: testPassword },
            });

        wrapper.find('form').simulate('submit', { preventDefault() {} });

        expect(startLoginEmailPassword).toHaveBeenCalled();
    });
});
