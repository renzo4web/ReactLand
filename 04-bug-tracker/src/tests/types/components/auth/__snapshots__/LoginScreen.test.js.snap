// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <LoginScreen /> wrapper should snapshot 1`] = `
Array [
  <h1>
    Login
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <div
      className="ui__input-group"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="test@gmail.com"
        type="text"
        value=""
      />
    </div>
    <div
      className="ui__input-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="auth__btn"
      disabled={true}
      type="submit"
    >
      Login
    </button>
    <hr />
    <div>
      <p>
        Login with social networks
      </p>
      <div
        className="google-btn"
        onClick={[Function]}
      >
        <div
          className="google-icon-wrapper"
        >
          <img
            alt="google button"
            className="google-icon"
            src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
          />
        </div>
        <p
          className="btn-text"
        >
          <b>
            Sign in with google
          </b>
        </p>
      </div>
    </div>
    <a
      href="/auth/register"
      onClick={[Function]}
    >
      Create new account
    </a>
  </form>,
]
`;
